dan_layer/core/src/workers/hotstuff_waiter.rs

```diff
@@ -74,6 +75,9 @@ use crate::{
 };
 
 const LOG_TARGET: &str = "tari::dan_layer::hotstuff_waiter";
+// When the foreign committee fails to provide QC, we ask f+1+EPSILON nodes for the QC (f+1 should be enough, but we
+// want to make sure, and we don't want to ask the whole committee).
```

Comment:
Do we gain any security guarantees by adding the `EPSILON` value ? A subset of `f+1` nodes should have at least one honest node inside, no ?

Reply:
Do we know the pubkeys of the foreign committee? If so, 1 should be enough. A QC holds the aggregated signature of 2f+1 votes for the given node. These can't be faked.

Reply:
You have to ask f + 1 to make sure you get an honest node. If you ask less than f, they could say they haven't gotten it